Reactive architecture

Kavignon opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gameoflightandshadows/a-game-of-light-and-shadows/107)
<!-- Reviewable:end -->

Commit:
#88 implemented basic version of the team party manager. The rest of the protocol needs the dynamic prototype to be completed before matching on those cases,.

Commit:
#82
The item store can receive the information from the team party manager.
The store can validate that the good mode was selected before doing any transaction !

Commit:
#82 -
Added getter and setters functions implemented as lens to generically pair them as a tuple when getting and equipment new equipment

Commit:
#82
Define an equipment system where players can update their character's equipment automatically after the purchase.

Commit:
#82 - Remove boiler plate code to auto equip purchase items bought directly from the store
If an equipment item is "drop", it'll move back to the inventory

Commit:
#82 -
Added a operationdatamodel in order to put only content relevant to the managers in the managers
Work begun for the battle system

review-ninja commented:
<a href="https://app.review.ninja/GameOfLightAndShadows/A-game-of-light-and-shadows/pull/107" target="_blank"><img src="https://app.review.ninja/66727011/pull/107/badge" alt="ReviewNinja"/>

Commit:
#82
For the battle system, it was required to differentiate human from brain controlled characters, which lead to some refactoring.

Commit:
#82 - Continue to refactor the code based on the changes on IGameCharacter, HumanCHaracter and BrainCharacter.
The inventory will be fixed soon.

Commit:
#82 -
Added the initialization messages for the battle system manager. The rest is incoming.

Commit:
#84 -
Either when the player stumbled on a treasure chest with money or when he defeats a brain character, he will acquired currency. If he sells stocks, he will acquire currency. He will lose money when he buys stock at the store.

Commit:
#82 - The battle system is complete.

Commit:
#108 Implemented partially the command manager.
The following requires an entire refactoring of the game item domain model which will happen now.

Commit:
#108 - Completely refactored the weaponry domain model to a more flexible model. There's less code duplication and weapon types such as different daggers are now interchangeable.

Kavignon commented:
the framework must be hand in. This branch must be merged for my supervisor.
